curp/src/server/cmd_worker/conflict_checked_mpmc.rs

```diff
@@ -297,32 +320,41 @@ impl<C: Command> Filter<C> {
                 ref cmd,
                 ref mut exe_st,
                 ref mut as_st,
-            } => match (*exe_st, *as_st) {
-                (ExeState::ExecuteReady, AsState::NotSynced | AsState::AfterSyncReady(_)) => {
+            } => match (*exe_st, as_st.clone()) {
+                (
+                    ExeState::ExecuteReady,
+                    AsState::NotSynced(prepare) | AsState::AfterSyncReady(_, prepare),
+                ) => {
+                    assert!(prepare.is_none(), "The prepare result of a given cmd can only be calculated when exe_state change from ExecuteReady to Executing");
+                    let mut prepare_err = None;
+                    match self.cmd_executor.prepare(cmd) {
+                        Ok(pre_res) => as_st.set_prepare_result(pre_res),
+                        Err(err) => prepare_err = Some(err.to_string()),
+                    }
```

Comment:
let prepare_err be the return value of the match expression, so that prepare_err can be immutable.

Reply:
Done